pkg/ccl/backupccl/key_rewriter.go

```diff
@@ -29,18 +29,28 @@ type prefixRewrite struct {
 	noop      bool
 }
 
+func (rewrite prefixRewrite) rewriteKey(key []byte) []byte {
```

Comment:
This function is in the _very_ hot path of regular restore, which already checked for noop in `(prefixRewriter) rewriteKey` before calling, it to avoid a function call, so we don't want to check it again here.

I don't know if we really care about it for online restores? A mem copy of a couple prefix bytes _per file_ doesn't really worry me? The special casing of noops seems worth doing for cases where it applies to _every single key-value_ in the entire restored data, but I'm less sure it is worth worrying about / adding extra `if`s when we're just messing with spans/metadata?

Reply:
I figured we likely wanted to avoid the function call. I don't feel stronly about my comment recommendation there.